### PR TITLE
TASK: Update to Neos 8.2

### PR DESCRIPTION
**Pull request by markusguenther**
Currently, this branch is used on staging.

**To-dos:**

- [ ] clear elastic indexes and reindex
- [x] sync assets from live
- [x] upgrade publish notifier (not 8.x compatible yet)
- [x] upgrade html wdiget (not 8.x compatible yet) or find another way (just 2 nodes use it)
- [x] misc

### CODE REVIEW

**Comment by markusguenther** [comment]
For the notifier, we just need to wait for @rolandschuetz to merge and release https://github.com/code-q-web-factory/neos-publishnotifier/pull/7

File: composer.json [hunk 1]
```diff
@@ -10,11 +10,10 @@
         }
     },
     "require": {
-        "php": "^7.4",
+        "php": "^8.1",
         "neos/docs-neos-io": "@dev",
 
-        "flownative/google-cloudstorage": "^5.0",
-        "rokka/imagine-vips": "^0.13"
```
Comment (kdambekalns) [inline composer.json:17]: Any specific reason for removing that one?

Comment (markusguenther) [reply composer.json:17]: Ah the moment it was just one reason... make the instance run ;)

But will readd that. Just needed a nap ;)

Comment (kdambekalns) [reply composer.json:17]: Ok. For https://github.com/neos/Neos.NeosIo/pull/470 I went with `^0.31`, hoping it'll work: https://github.com/neos/Neos.NeosIo/pull/470/commits/9dc23c9fd582c6b3853b07e61021f42e0b2f1e89#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R23

**Comment by kdambekalns** [comment]
Uh, `.warmupdone` should be removed and put into `.gitignore` as to not possibly confuse (real) Beach…

**Comment by markusguenther** [comment]
The HTML widget package is only used for this column. So we don't re-add this and just use content for that.
Screenshot 2022-11-30 at 12 02 36

**Comment by manuelmeister** [comment]
I use it for the fusionpen examples in #95. But I think this could be solved another way.

**Comment by markusguenther** [comment]
Ok here we have the same issue ... need to wait for Neos 8.2.2
https://github.com/neos/neos-development-collection/pull/3969

**Comment by rolandschuetz** [comment]
My packages which blocked the PR are updated.

**Comment by manuelmeister** [comment]
Why is the hypenation not working anymore on home with this PR? 
https://instance-6ee89105-ec8e-4024-97bb-5c2b9d49a15c.flownative.cloud/
Bildschirmfoto 2023-01-05 um 10 41 24

**Comment by kdambekalns** [comment]
> Why is the hypenation not working anymore on home with this PR?

🤷‍♂️ https://github.com/neos/Neos.DocsNeosIo/pull/100/commits/d0988656ea41ec4ad80a328575ddc9b0f0143c1b seems innocent enough…

**Comment by manuelmeister** [comment]
The package does something totally different in the new version: https://github.com/CarbonPackages/Carbon.Hyphen/pull/7

`||` is not replaced anymore. @jonnitto what is the migration path for the version update?

**Comment by kdambekalns** [comment]
How often is `||` actaully in the content?

**Comment by mhsdesign** [comment]
> I think it would be best to replace || with the unicode shy character and install [Shel.Neos.Hyphens](https://github.com/Sebobo/Shel.Neos.Hyphens) as suggested by Roland.

this is the way ;) +1

**Comment by Sebobo** [comment]
* Updates dependencies
* Reinstalled Rolands packages (HTMLWidget and Notifier)
* I replaced the hyphens stuff and enabled it on the silhouettes and document title.
* I fixed the compression character issue on the search results page (experienced that locally).
* Recompiled the CodeEditor plugin to make the backend work properly

So we can click around the stage again and check whether we are done here now.

**Review by Sebobo — APPROVED** [review]
I fixed all issues I have found

**Comment by markusguenther** [comment]
Thank you very much to all of you 💙

**Comment by markusguenther** [comment]
I have added a small migration for the replacement of `||` to the soft hyphens. We just use it in the title property and somewhere in Code, but in the Code node types I did not replace it, of course.

From my side we can merge it. Tested it locally and on staging.

**Pull request merged**